components/file_attachment/file_attachment.tsx

```diff
@@ -121,6 +139,122 @@ export default class FileAttachment extends React.PureComponent<Props, State> {
         }
     }
 
+    refCallback = (menuRef: Menu) => {
+        if (menuRef) {
+            const anchorRect = this.buttonRef.current?.getBoundingClientRect();
+            let y;
+            if (typeof anchorRect?.y === 'undefined') {
+                y = typeof anchorRect?.top == 'undefined' ? 0 : anchorRect?.top;
+            } else {
+                y = anchorRect?.y;
+            }
+            const windowHeight = window.innerHeight;
+
+            const totalSpace = windowHeight - 80;
+            const spaceOnTop = y - Constants.CHANNEL_HEADER_HEIGHT;
+            const spaceOnBottom = (totalSpace - (spaceOnTop + Constants.POST_AREA_HEIGHT));
+
+            this.setState({
+                openUp: (spaceOnTop > spaceOnBottom),
+            });
+        }
+    }
+
+    private handleDropdownOpened = (open: boolean) => {
+        this.props.handleFileDropdownOpened(open);
+        this.setState({keepOpen: open});
+    }
+
+    handleGetPublicLink = () => {
+        GlobalActions.showGetPublicLinkModal(this.props.fileInfo.id);
+    }
+
+    private renderFileMenuItems = () => {
+        const {enablePublicLink, fileInfo, pluginMenuItems} = this.props;
+
+        let divider;
+        const defaultItems = [];
+        if (enablePublicLink) {
+            defaultItems.push(
+                <Menu.ItemAction
+                    data-title='Public Image'
+                    onClick={this.handleGetPublicLink}
+                    ariaLabel={localizeMessage('view_image_popover.publicLink', 'Get a public link')}
+                    text={localizeMessage('view_image_popover.publicLink', 'Get a public link')}
+                />,
+            );
+        }
+
+        const pluginItems = pluginMenuItems?.filter((item) => item?.match(fileInfo)).map((item) => {
+            return (
+                <Menu.ItemAction
+                    id={item.id + '_pluginmenuitem'}
+                    key={item.id + '_pluginmenuitem'}
+                    onClick={() => item?.action(fileInfo)}
+                    text={item.text}
+                />
+            );
+        });
+
+        if (!defaultItems?.length && !pluginItems?.length) {
```

Comment:
condition can be simplified
```suggestion
        if (!(defaultItems?.length || pluginItems?.length)) {
```

Reply:
done